lib/inspec/profile_vendor.rb

```diff
@@ -80,5 +82,40 @@ def extract_archives
         File.delete(filepath)
       end
     end
+
+    # Check top-level profile metadata for any local archives
+    # if present, go ahead and inflate it into the vendor cache.
+    # This will cause subsequent fetchers to hit on the vendor cache,
+    # avoiding a possibly failing fetch under airgapped conditions.
+    def warm_vendor_cache_from_archives
+      # We need to determine the cwd of the profile, to expand the (likely)
+      # relative path of the dependencies. There is no non-invasive way of doing that.
+      profile_cwd = profile.source_reader.target.parent.path
+
+      # list dependencies in profile
+      profile.metadata.dependencies.each do |dep_info|
+        next unless dep_info.key?(:path) # "Local"-type fetchers are identified by the :path key
+
+        # Locate local dep
+        dep_profile_path = File.expand_path(dep_info[:path], profile_cwd)
+        raise(Inspec::FetcherFailure, "Profile dependency local path '#{dep_profile_path}' does not exist") unless File.exist?(dep_profile_path)
+
+        # Determine if it is an archive
+        file_provider = FileProvider.for_path(dep_profile_path)
+        next unless file_provider.is_a?(ZipProvider) || file_provider.is_a?(TarProvider)
```

Comment:
If the fetcher system is plugable, then we have to update this list dynamically.

Reply:
Agreed. Typechecks are a smell.

Both `ZipProvider` and `TarProvider` subclass from `FileProvider`... If they moved to subclassing `ArchiveProvider` (which subclassed `FileProvider`) then only one check needs to be made here and it is more resilient to change.

Reply:
This is the `FileProvider` system, not the `Fetcher`, as I understand it, and is used in other places to detect whether a file is an archive - in fact, just a few lines above at https://github.com/inspec/inspec/blob/8b5789004ac31f4b6e7210421b119af41f483e3e/lib/inspec/profile_vendor.rb#L74 . Good idea or not, "how do I detect an archive in InSpec" has at least one existing answer - and it is based on typechecking. Sounds like you'd like a  different approach, and you're like me to alter an unrelated class hierarchy to avoid a code smell.

Respectfully, I feel that is outside the scope of this PR. I'll open an issue.